account.go

```diff
@@ -46,59 +53,39 @@ func (sc StarkCurve) NewSigner(private, pubX, pubY *big.Int, gw *StarknetGateway
 	- implementation has been tested against OpenZeppelin Account contract as of: https://github.com/OpenZeppelin/cairo-contracts/blob/4116c1ecbed9f821a2aa714c993a35c1682c946e/src/openzeppelin/account/Account.cairo
 	- accepts a multicall
 */
-func (signer *Signer) Execute(ctx context.Context, address string, txs []Transaction) (addResp AddTxResponse, err error) {
-	nonce, err := signer.Gateway.AccountNonce(ctx, address)
+func (signer *Signer) Execute(ctx context.Context, address string, txs []types.Transaction) (*types.AddTxResponse, error) {
+	nonce, err := signer.provider.AccountNonce(ctx, address)
 	if err != nil {
-		return addResp, err
+		return nil, err
 	}
 
 	maxFee := big.NewInt(0)
+	chainID, err := signer.provider.ChainID(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-	hash, err := signer.Curve.HashMulticall(address, nonce, maxFee, UTF8StrToBig(signer.Gateway.ChainId), txs)
+	hash, err := signer.Curve.HashMulticall(address, nonce, maxFee, HexToBN(chainID), txs)
```

Comment:
This will need to be flipped back to `UTF8StrToBig(`

Reply:
Ah my bad will switch it back!